subprojects/core/src/main/java/org/gradle/api/internal/artifacts/publish/DefaultPublishArtifact.java

```diff
@@ -106,4 +107,27 @@ public void setDate(Date date) {
     public void setFile(File file) {
         this.file = file;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I don't think we should have `equals()` or `hashCode()` implementations on this type, particularly implementations that uses mutable values. These domain object should just be buckets of data and we should validate/deduplicate later based on how the set of artifacts is used.

What do we need these for?